docker/archive/dest.go

```diff
@@ -19,22 +19,28 @@ func newImageDestination(ctx *types.SystemContext, ref archiveReference) (types.
 	if ref.destinationRef == nil {
 		return nil, errors.Errorf("docker-archive: destination reference not supplied (must be of form <path>:<reference:tag>)")
 	}
-	fh, err := os.OpenFile(ref.path, os.O_WRONLY|os.O_EXCL|os.O_CREATE, 0644)
+
+	// the file needs to be created for write only
+	// if the file prexisted it needs to be empty
+	fh, err := os.OpenFile(ref.path, os.O_WRONLY|os.O_CREATE, 0644)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error opening file %q", ref.path)
+	}
+
+	fhStat, err := fh.Stat()
 	if err != nil {
-		// FIXME: It should be possible to modify archives, but the only really
-		//        sane way of doing it is to create a copy of the image, modify
-		//        it and then do a rename(2).
-		if os.IsExist(err) {
-			err = errors.New("docker-archive doesn't support modifying existing images")
-		}
-		return nil, err
+		return nil, errors.Wrapf(err, "error statting file %q", ref.path)
+	}
+
+	if fhStat.Mode()&os.ModeNamedPipe != 0 || (fhStat.Mode().IsRegular() && fhStat.Size() == 0) {
```

Comment:
Please use `(fhStat.Mode() & os.ModeType) == os.ModeNamedPipe`.  The code above _is_ correct (unlike UNIX, `os.FileMode` uses individual bits for the various non-regular-file file types), but unexpectedly tricky because the negation of the condition is not `… == 0` (that would also include regular files). 

(Also, as a matter of personal? style, `if` (the error case) `{` return error `}` /* normal processing */ would be more idiomatic.)